src/lib.rs

```diff
@@ -160,15 +175,79 @@ impl Platform {
                         touch.location.y as f32 / self.scale_factor as f32,
                     );
 
-                    self.raw_input.events.push(egui::Event::PointerButton {
-                        pos: pointer_pos,
-                        button: egui::PointerButton::Primary,
-                        pressed: match touch.phase {
-                            winit::event::TouchPhase::Started => true,
-                            _ => false,
+                    let device_id = match self.device_indices.get(&touch.device_id) {
+                        Some(id) => *id,
+                        None => {
+                            self.device_indices.insert(touch.device_id, self.next_device_index);
+                            self.next_device_index += 1;
+                            *self.device_indices.get(&touch.device_id).unwrap()
                         },
-                        modifiers: Default::default(),
+                    };
+                    let egui_phase = match touch.phase {
+                        TouchPhase::Started => egui::TouchPhase::Start,
+                        TouchPhase::Moved => egui::TouchPhase::Move,
+                        TouchPhase::Ended => egui::TouchPhase::End,
+                        TouchPhase::Cancelled => egui::TouchPhase::Cancel,
+                    };
+
+                    let force = match touch.force {
+                        Some(winit::event::Force::Calibrated { force, .. }) => force as f32,
+                        Some(winit::event::Force::Normalized(force)) => force as f32,
+                        None => 0.0f32 // hmmm, egui can't differentiate unsupported from zero pressure
+                    };
+
+                    self.raw_input.events.push(egui::Event::Touch {
+                        device_id: egui::TouchDeviceId(device_id),
+                        id: egui::TouchId(touch.id),
+                        phase: egui_phase,
+                        pos: pointer_pos,
+                        force
                     });
+
+                    // Currently Winit doesn't emulate pointer events based on
+                    // touch events but Egui requires pointer emulation...
+                    //
+                    // For simplicity we just merge all touch pointers into a
+                    // single virtual pointer and ref-count the press state
+                    // (i.e. the pointer will remain pressed during multi-touch
+                    // events until the last pointer is lifted up)
+
+                    let was_pressed = self.touch_pointer_pressed > 0;
+
+                    match touch.phase {
+                        TouchPhase::Started => {
+                            self.touch_pointer_pressed += 1;
+                        },
+                        TouchPhase::Ended | TouchPhase::Cancelled => {
+                            self.touch_pointer_pressed -= 1;
```

Comment:
This screams for checked_sub. Then we can also use u32 for the "touch_pointer_pressed".